### PR TITLE
feat!: add --exclude and --include for list

### PR DESCRIPTION
**Pull request by blacha**
--exclude takes priority if something is excluded it will no be included

this removes --filter in favor of --include/--exclude

### CODE REVIEW

File: src/commands/list/__test__/filter.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { fsa } from '@chunkd/fs';
+import o from 'ospec';
+import { asyncFilter } from '../list.js';
+
+o.spec('AsyncFilter', () => {
+  function makGenerator(list: string[]): () => AsyncGenerator<{ path: string }> {
```
Comment (paulfouquet) [inline src/commands/list/__test__/filter.test.ts:6]: What means `mak`?

Comment (blacha) [reply src/commands/list/__test__/filter.test.ts:6]: haha `make` Ill fix it

File: src/commands/list/list.ts [hunk 2]
```diff
@@ -58,14 +58,15 @@ export const commandList = command({
 
 export async function* asyncFilter<T extends { path: string }>(
   source: AsyncGenerator<T>,
-  filter: string,
+  opts?: { include?: string; exclude?: string }, //filter: string,
```
Comment (paulfouquet) [inline src/commands/list/list.ts:61]: Is `//filter: string,` intentionally being left?

Comment (blacha) [reply src/commands/list/list.ts:61]: Nope!

**Review by paulfouquet — APPROVED** [review]

**Pull request merged**